system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,98 +28,186 @@ The application package has the following folder structure:
+| site             | string          | N               | Link to the application's website.|
+| tagline          | string          | N               | The application's slogan.|
+
+**Author Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | N               | The name of the application's creator.|
+| email            | string          | N               | Email address of the application's creator.|
+
+**Organization Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | Y               | Organization responsible for the application's development and distribution.|
+| site             | string          | N               | Link to the organization's website.|
+
+**Source Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| cluster          | []SourceType    | N               | Source type element for indicating the source for deploying the application on a cluster using Helm. See the [Source Type](#source-type-attributes) section below. |
+| stand-alone      | []SourceType    | N               | Source type element for indicating the source for deploying the application on a stand-alone device using Docker compose. See the [Source Type](#source-type-attributes) section below. |
+
+**Source Type Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | Y               | A unique name used to identify the package source. For helm installations the name will be used as the chart name. |
+| type             | string          | Y               | Indicates the source's package format. The values are `helm.v3` to indicate the source's package format is Helm version 3 and `docker-compose` to indicate the source's package format is Docker Compose. When installing the application on a device supporting the Kubernetes platform all `helm.v3` sources, and only `helm.v3` sources, will be provided to the device in same order they are listed in the application description file. When installing the application on a device supporting docker-compose all `docker-compose` sources, and only `docker-compose` sources, will be provided to the device in the same order they are listed in the application description file. The device will install the sources in the same order they are listed in the application description file. Source types under `kubernetes` must use `helm.v3`. Source types under `moby` must use `docker-compose` |
```

Comment:
'kubernetes'  -> 'cluster' and 'moby' -> 'stand-alone' for consistency. And if 'cluster' and 'stand-alone' already decides the source type to use, mabybe we don't need source type here.

Reply:
The idea here is in the future we may have different deployment pacakge types (e.g., web assembly or helm v4) that can be deployed to the cluster but have different properties or need additional tooling on the device to install that package type. I can see in the future it being possible to deploy other package types to the cluster than helm charts. Some goes with the stand-alone devices. Right now we're talking about docker-compose but in the future we might support some other package type. 

So for situations like this do you feel there is a better approach to handling this?

Reply:
Fixed the missed edits after the last set of changes.